src/Entity/DispatchingEntityIdParser.php

```diff
@@ -27,6 +27,10 @@ class DispatchingEntityIdParser implements EntityIdParser {
 	 * @param callable[] $idBuilders
 	 */
 	public function __construct( array $idBuilders ) {
+		if ( $this->idBuilders === [] ) {
+			throw new InvalidArgumentException( '$idBuilders must not be empty' );
```

Comment:
Uh. I find this a little problematic. I believe this was done as it was before for a reason: The idea was to allow a setup with *no* custom entity types. As long as the parse method is not called, it is not a problem to have an empty ID parser.

Even if such a situation can never happen in production, this change makes this entire patch a breaking one. We could avoid this by keeping the previous handling of empty arrays.

The least minimum is to documented this as a `@throws`.

Reply:
Moved back, this probably hardly has any performance impact. It mostly looked weird to me.